pkg/proxy/iptables/proxier.go

```diff
@@ -927,7 +927,7 @@ func (proxier *Proxier) syncProxyRules() {
 	}
 
 	// Accumulate NAT chains to keep.
-	activeNATChains := map[utiliptables.Chain]bool{} // use a map as a set
+	activeNATChains := sets.New[utiliptables.Chain]()
```

Comment:
I don't think this change should be backported, adding logs or metrics is no risk and improves monitoring, it is totally justified
However, this commit is a refactor and has risk of regression, it also not fixing a bug